index.bs

```diff
@@ -403,107 +410,100 @@ these steps:
+     If |position| is past the end of the file writing will fail and this method rejects.
 
      There are no guarantees that the data written to the file using this method is actually
-     reflected in the file on disk until the {{FileSystemWriter/close()}}
-     method is called. Specifically when |writer|'s
+     reflected in the file on disk until the {{FileSystemWritableFileStream/close()}}
+     method is called. Specifically when |writableFStream|'s
      [=atomic flag=] is set, the user agent will ensure that no changes are written to the actual
      file until one of those methods is called.
 </div>
 
 <div algorithm>
-The <dfn method for=FileSystemWriter>asWritableStream()</dfn> method, when invoked,
-must run these steps:
+The <dfn method for=FileSystemWritableFileStream>write(|position|, |data|)</dfn> method, when invoked, must run
+these steps:
 
 1. TODO
 
 </div>
 
-### The {{FileSystemWriter/truncate()}} method ### {#api-filesystemwriter-truncate}
+### The {{FileSystemWritableFileStream/truncate()}} method ### {#api-filesystemwritablefilestream-truncate}
 
 <div class="note domintro">
-  : await |writer| . {{FileSystemWriter/truncate()|truncate}}(|size|)
-  :: Resizes the file associated with |writer| to be |size| bytes long. If |size| is larger than
+  : await |writableFStream| . {{FileSystemWritableFileStream/truncate()|truncate}}(|size|)
+  :: Resizes the file associated with |writableFStream| to be |size| bytes long. If |size| is larger than
      the current file size this pads the file with zero bytes, otherwise it truncates the file.
 
      There are no guarantees that the changes made to the file using this method are actually
-     reflected in the file on disk until the {{FileSystemWriter/close()}}
-     method is called. Specifically when |writer|'s
+     reflected in the file on disk until the {{FileSystemWritableFileStream/close()}}
+     method is called. Specifically when |writableFStream|'s
      [=atomic flag=] is set, the user agent will ensure that no changes are written to the actual
      file until one of those methods is called.
 </div>
 
 <div algorithm>
-The <dfn method for=FileSystemWriter>truncate(|size|)</dfn> method, when invoked, must run these
+The <dfn method for=FileSystemWritableFileStream>truncate(|size|)</dfn> method, when invoked, must run these
 steps:
 
 1. TODO
 
 </div>
 
-### The {{FileSystemWriter/close()}} method ### {#api-filesystemwriter-close}
+### The {{FileSystemWritableFileStream/close()}} method ### {#api-filesystemwritablefilestream-close}
 
 <div class="note domintro">
-  : await |writer| . {{FileSystemWriter/close()}}
-  :: First flushes any data written so far to disk, and then closes the writer.
-     If |writer|'s [=atomic flag=] is set, no changes will
+  : await |writableFStream| . {{FileSystemWritableFileStream/close()}}
+  :: This method is similar to {{WritableStreamDefaultWriter/close()|WritableStreamDefaultWriter's close()}}
```

Comment:
should just calling close() on the Writer have the same behavior? I imagine it should, as otherwise it would be kind of weird that the only way to actually get your changes made using a writer to persist on disk would be to first call `releaseLock` on the writer, and then call close on the stream. (I at least imagine we would want write/truncate/close on the stream to only work when the stream isn't locked, and to itself keep the stream locked during the duration of the operation).

Reply:
I think it should.

Reply:
See my above comments for how to achieve that, by defining underlying sink algorithms.

Reply:
Behavior now defined in WritableStream.